lsr_role2collection.py

```diff
@@ -284,17 +287,32 @@ def file_replace(path, find, replace, file_patterns):
         "default to '_'"
     ),
 )
-args = parser.parse_args()
+args, unknown = parser.parse_known_args()
+
+role = args.role
+if not role:
```

Comment:
What happens if you specify only `--molecule`?

Reply:
Yeah, I was testing it and ran into the problem :p.
And I'm going to rename `--molecule` with `--tox` since it's more accurate...
I also found if I run `tox -e pylint`, it fails to find out the modules to import - both custom ones and official ones. It seems I need to set some env var to tell something to pylint?
Instead, I used ansible-test command to launch pylint, but it still fails badly...  Keep invenstigating...
```
ERROR: Found 174 pylint issue(s) which need to be resolved:
ERROR: plugins/module_utils/certificate/certificate/providers/base.py:283:31: ansible-format-automatic-specification: Format string contains automatic field numbering specification
ERROR: plugins/module_utils/certificate/certificate/providers/base.py:419:16: ansible-format-automatic-specification: Format string contains automatic field numbering specification
...
```

Reply:
hmm - I'm not seeing what you are seeing - I'm running `tox` with your latest code:
```
tox 
black installed: appdirs==1.4.4,black==20.8b1,click==7.1.2,mypy-extensions==0.4.3,pathspec==0.8.0,regex==2020.7.14,toml==0.10.1,typed-ast==1.4.1,typing-extensions==3.7.4.3
black run-test-pre: PYTHONHASHSEED='3596654172'
...
pylint installed: ansible==2.9.13,astroid==2.4.2,cffi==1.14.2,colorama==0.4.3,cryptography==3.1,isort==5.5.2,Jinja2==2.11.2,lazy-object-proxy==1.4.3,MarkupSafe==1.1.1,mccabe==0.6.1,pycparser==2.20,pylint==2.6.0,PyYAML==5.3.1,ruamel.yaml==0.16.12,ruamel.yaml.clib==0.2.2,six==1.15.0,toml==0.10.1,wrapt==1.12.1
pylint run-test-pre: PYTHONHASHSEED='3596654172'
pylint run-test: commands[0] | bash -c 'find \( -name .tox -prune \) -o \( -name .venv -prune \) -o -name \*.py -exec pylint --errors-only  {} \;'
...
  pylint: commands succeeded
```

Reply:
Whoa... What's the difference...??? I thought it might be from the location of the converted collections. I'm usually using non-standard place like /tmp/collections. But this time I placed the converted collections in  ~/.ansible/collections, but my `tox -e pylint` still fails... (;_;)
```
************* Module modules.certificate_request
E:279, 0: No name 'certificate' in module 'ansible.module_utils' (no-name-in-module)
E:279, 0: Unable to import 'ansible.module_utils.certificate.providers' (import-error)
************* Module modules.bsize
E: 55, 0: Unable to import 'ansible_collections.fedora.system_roles.plugins.module_utils.storage.size' (import-error)
************* Module modules.kernel_settings
E: 18, 0: Unable to import 'configobj' (import-error)
E:563, 4: Unable to import 'pyudev.monitor' (import-error)
```

Reply:
Maybe, this is my problem... My python is 3.8.5 and I have pylint 2.6. Why I'm getting this DEPRECATION message???
```
pylint installed: DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support,ansible==2.9.13,astroid==1.6.6,backports.functools-lru-cache==1.6.1,cffi==1.14.2,colorama==0.4.3,configparser==4.0.2,cryptography==3.1,enum34==1.1.10,futures==3.3.0,ipaddress==1.0.23,isort==4.3.21,Jinja2==2.11.2,lazy-object-proxy==1.5.1,MarkupSafe==1.1.1,mccabe==0.6.1,pycparser==2.20,pylint==1.9.5,PyYAML==5.3.1,singledispatch==3.4.0.3,six==1.15.0,wrapt==1.12.1
pylint run-test-pre: PYTHONHASHSEED='2535144570'
```

Reply:
No, it's not likely. I forced to use python 3.8 (I hope), but it still fails in the same way...
```
$ TRAVIS_PYTHON_VERSION=3.8 tox -e pylint
pylint recreate: /home/nhosoi/.ansible/collections/ansible_collections/fedora/system_roles/.tox/env-3.8
pylint installdeps: ansible, colorama, pylint>=1.8.4, -rpylint_extra_requirements.txt
pylint installed: ansible==2.9.13,astroid==2.4.2,cffi==1.14.2,colorama==0.4.3,cryptography==3.1,isort==5.5.2,Jinja2==2.11.2,lazy-object-proxy==1.4.3,MarkupSafe==1.1.1,mccabe==0.6.1,pycparser==2.20,pylint==2.6.0,PyYAML==5.3.1,six==1.15.0,toml==0.10.1,wrapt==1.12.1
pylint run-test-pre: PYTHONHASHSEED='2696532288'
pylint run-test: commands[0] | bash /home/nhosoi/.ansible/collections/ansible_collections/fedora/system_roles/.travis/runpylint.sh --errors-only
<<snip>>
************* Module modules.certificate_request
plugins/modules/certificate_request.py:279:0: E0611: No name 'certificate' in module 'ansible.module_utils' (no-name-in-module)
plugins/modules/certificate_request.py:279:0: E0401: Unable to import 'ansible.module_utils.certificate.providers' (import-error)
<<snip>>
```

Reply:
the tox.ini in auto-maintenance doesn't use `TRAVIS_PYTHON_VERSION`.  The python version is set here:
```
[testenv]
passenv = *
basepython = python3
```
Try removing your tox environments and recreating them: `rm -rf .tox` or `tox -r`

re: But this time I placed the converted collections in ~/.ansible/collections

If you are going to run pylint and pytest against all of the other modules, you'll need to install all of their dependencies.  But you shouldn't be doing that for working on just the auto-maintenance code.  The goal initially is that `tox` will run `pylint` and `pytest` against the python files only in the auto-maintenance directory.

Yes, we will eventually need some sort of tox/travis based CI to run against the converted collections, but that is a different task.

Reply:
Thanks for your comments, @richm. Yes, running `tox` in auto-maintenance is successful.

Do you think I could stop this pr here? BTW, this pr is made from 32 commits now... Do you recommend me to squash them into several commits which has some meaningful descriptions per each?

Reply:
> Do you think I could stop this pr here?

Yes.  Let's start using this to develop tox/travis/integration CI for collections.

I would like either @pcahyna or @t-woerner to review this - perhaps they will have a preference about how they would like the commits squashed, if any, to help them review.

This should be squashed into a single commit before merging though.